packages/block-editor/src/components/use-setting/index.js

```diff
@@ -49,6 +49,18 @@ const deprecatedFlags = {
 	'spacing.customPadding': ( settings ) => settings.enableCustomSpacing,
 };
 
+const filterColorsFromCoreOrigin = ( path, colors ) => {
```

Comment:
This logic of just excluding core colors causes a bug. If using the global styles color palette editor as a user I add a new color to the existing theme palette. Instead of just the new color appearing, all the other colors appear two times:
![image](https://user-images.githubusercontent.com/11271197/118718505-39e8ff00-b81f-11eb-93cb-bd8da57728bf.png)


Maybe instead of just excluding core colors from the UI, we should have a hierarchy of origins core -> theme -> child theme 1 -> child theme 2 ... -> user and the UI for now just show the higher colors from the hierarchy.

Reply:
I can't repro this one, tried a bunch of scenarios but it works for me:

https://user-images.githubusercontent.com/583546/118780396-4b1c2500-b88c-11eb-95fc-1360d648caf8.mp4

Reply:
This was a hard one. I forgot to consider colors, so thanks for bringing that up. It's now fixed.